docs/modules/clients/pages/java.adoc

```diff
@@ -862,14 +917,120 @@ Programmatic Configuration:
 ----
 ClientConfig clientConfig = new ClientConfig();
 ClientNetworkConfig networkConfig = clientConfig.getNetworkConfig();
-networkConfig.setSmartRouting(true);
+networkConfig.setClusterRoutingMode(ALL_MEMBERS);
 ----
 
-Its default value is `true` (smart client mode).
+**SINGLE_MEMBER**
+
+To connect to a single member, which can be used as a gateway to the other members, use the `SINGLE_MEMBER` cluster routing mode, which can be defined as described below.
 
-Note that you need to disable smart routing (`false`) for the clients which
-want to use temporary permissions defined in a member.
-See the xref:security:native-client-security.adoc#handling-permissions-when-a-new-member-joins[Handling Permissions section].
+When using the `SINGLE_MEMBER` cluster routing mode, consider the following:
+
+* The absence of <<configuring-backup-acknowledgement, backup acknowledgements>>, as the client does not have a view of the entire cluster
+* If you have multiple members on a single machine, we advise that <<configuring-address-list,explicit ports are set for each member>>
+* Connections to the xref:cp-subsystem:cp-subsystem.adoc[CP Subsystem] are direct-to-leader, which can result in improved performance
+* <<configuring-load-balancer,`LoadBalancer`>> configuration is ignored
+
+Declarative Configuration:
+
+[tabs] 
+==== 
+XML:: 
++ 
+-- 
+[source,xml]
+----
+<hazelcast-client>
+    ...
+    <network>
+        </cluster-routing mode="SINGLE_MEMBER">
+    </network>
+    ...
+</hazelcast-client>
+----
+--
+
+YAML::
++
+[source,yaml]
+----
+hazelcast-client:
+  network:
+    cluster-routing:
+      mode: SINGLE_MEMBER
+----
+====
+
+Programmatic Configuration:
+
+[source,java]
+----
+ClientConfig clientConfig = new ClientConfig();
+ClientNetworkConfig networkConfig = clientConfig.getNetworkConfig();
+networkConfig.setClusterRoutingMode(SINGLE_MEMBER);
+----
+
+**MULTI_MEMBER**
+
+To connect to a subset partition grouping, which allows direct connection to the specified group and gateway connections to other members, use the `MULTI_MEMBER` cluster routing mode, which can be defined as follows.
+
+To use the `MULTI-MEMBER` cluster routing mode, you must also define the grouping strategy to apply. For further information on configuring partition groups, see xref:clusters:partition-group-configuration.adoc[]. 
```

Comment:
```suggestion
To use the `MULTI_MEMBER` cluster routing mode, you must also define the grouping strategy to apply. For further information on configuring partition groups, see xref:clusters:partition-group-configuration.adoc[]. 
```

Reply:
Alongside my previous comment about whether or not we should discuss the possibility of new grouping strategies in the future, the default value will be `PARTITION_GROUPS` so long as it's the only option available, so if we do decide we don't want to talk about the future possibilities, maybe we can exclude the grouping strategy for simplicity? cc @gareth-johnston wdyt?

Reply:
In a previous comment we agreed that "coming soon" is marketing's job - given that `MULTI_MEMBER` only supports partition group (which is the default value), maybe we simplify the docs here and only talk about enabling with `MULTI_MEMBER`, and ignore the `routing-strategy` in docs until it becomes relevant?